File: packages/shared/src/types/vibe.ts
// Mirrors packages/shared/schemas/vibe-profile.schema.json
// and apps/api/app/schemas/vibe.py — keep all three in sync.

export type Mood =
  | "happy"
  | "sad"
  | "calm"
  | "anxious"
  | "angry"
  | "excited"
  | "romantic"
  | "nostalgic"
  | "neutral";

export type Energy = "very_low" | "low" | "medium" | "high" | "very_high";

export type Valence = "negative" | "neutral" | "positive";

export type VibeContext = "morning" | "afternoon" | "evening" | "night" | "unknown";

export type SocialEnergy = "low" | "medium" | "high";

export interface VibeProfile {
  mood: Mood;
  energy: Energy;
  valence: Valence;
  context: VibeContext;
  social_energy: SocialEnergy;
  vibe_tags: string[];
  confidence: number;
}

File: apps/api/app/schemas/vibe.py
"""API request/response contracts for the vibe endpoint.

Mirrors the shape defined in packages/shared/schemas/vibe-profile.schema.json
— keep the two in sync.
"""

from pydantic import BaseModel


class VibeRequest(BaseModel):
    text: str
    preferences: dict | None = None
    # Optional photo is accepted as a separate multipart field on the route,
    # not as part of this JSON body — see app/api/routes/vibe.py.


class VibeProfile(BaseModel):
    mood: str
    energy: str
    valence: str
    context: str
    social_energy: str
    vibe_tags: list[str]
    confidence: float
